src/corecel/sys/ScopedProfiling.nvtx.cc

```diff
@@ -0,0 +1,113 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/sys/ScopedProfiling.nvtx.cc
+//---------------------------------------------------------------------------//
+
+#include "ScopedProfiling.hh"
+
+#include <mutex>
+#include <shared_mutex>
+#include <unordered_map>
+#include <nvtx3/nvToolsExt.h>
+
+#include "celeritas_config.h"
+#include "corecel/Assert.hh"
+#include "corecel/io/Logger.hh"
+
+namespace celeritas
+{
+namespace
+{
+
+//---------------------------------------------------------------------------//
+/*!
+ * Global registry for strings used by NVTX.
+ * This Implemented as a free function instead of a static in
+ * ScopedProfiling to hide the NVTX dependency from user of the
+ * interface.
+ */
+std::unordered_map<std::string, nvtxStringHandle_t>& message_registry()
+{
+    static std::unordered_map<std::string, nvtxStringHandle_t> registry;
+    return registry;
+}
+
+nvtxDomainHandle_t domain_handle()
+{
+    static nvtxDomainHandle_t domain = nvtxDomainCreateA("celeritas");
+    return domain;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Retrieve the handle for a given message, insert it if it doesn't already
+ * exists
+ */
+nvtxStringHandle_t message_handle_for(std::string const& message)
+{
+    static std::shared_mutex mutex;
+
+    {
+        std::shared_lock lock(mutex);
+        auto message_handle = message_registry().find(message);
+        if (message_handle != message_registry().end())
+        {
+            return message_handle->second;
+        }
+    }
+    // we did not find the handle, insert it
```

Comment:
Try to use leading case for comments
```suggestion
    // We did not find the handle: try to insert it
```